internal/fxcontext/context.go

```diff
@@ -58,15 +58,28 @@ func New(ctx gcontext.Context, host service.Host) fx.Context {
 // Logger returns context based logger. If logger is absent from the context,
 // the function updates the context with a new context based logger
 func (c *Context) Logger() ulog.Log {
-	return c.getStore().log
+	store := c.getStore()
+	if store.log == nil {
```

Comment:
Store can be created without a logger. That doesn't make much sense right now but in future once we add more things to the store object for eg. auth etc. It is possible to have one set and not the other.